corda/corda
# [EG-438] Error Reporting Framework

This PR introduces the error reporting framework, which is a small piece of code to allow errors to be reported to the logs in a more robust way. It also allows for these errors to be internationalized using resource bundles. Note that in order for an error to be reported, it has to be marked in the code with the `ErrorCode` interface and reported by calling `report` on a logger. This code may be pulled into its own library at some point, and so the code aims to have as few dependencies on other parts of Corda as possible.

Some small changes are made in the Corda code to start the library on node startup, and to introduce a small number of error conditions. This will be improved on in future commits to this feature branch.

A utility is also included for generating error code documentation for hosting on the docs site, using the resource bundles. This is included in `tools/error-page-builder`.

## Testing
- Check that errors are correctly reported in the initially supported locale (English)
- Check that the error page builder generates a valid markdown table using different locales.
- Check the error page builder handles invalid directories and the file to generate already existing

(There's still some testing to do around subcommands of the node.)
